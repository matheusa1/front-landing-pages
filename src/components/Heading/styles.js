import styled, { css } from 'styled-components'

const titleSize = {
  small: (theme) => css`
    font-size: ${theme.font.sizes.md}
  `,
  medium: (theme) => css`
    font-size: ${theme.font.sizes.lg}  
  `,
  big: (theme) => css`
    font-size: ${theme.font.sizes.xlg}
  `,
  huge: (theme) => css`
    font-size: ${theme.font.sizes.xhg}
  `,
}

const titleCase = (uppercase) => {
  return css`
    text-transform: ${uppercase ? 'uppercase' : 'none'};
  `
}

export const Title = styled.h1`
  ${({ theme, colorDark, size, uppercase }) => css`
    color: ${colorDark ? theme.colors.primaryColor : theme.colors.white};
    ${titleSize[size](theme)};
    ${titleCase(uppercase)};
  `}
`
