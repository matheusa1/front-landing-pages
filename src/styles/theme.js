export const theme = {
  colors: {
    primaryColor: '#0a1128',
    secondaryColor: '#00F',
    white: '#f1f1f1',
  },
}
