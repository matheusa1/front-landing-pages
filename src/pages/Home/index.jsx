import React from 'react'

import Heading from '../../components/Heading'
import * as S from './styles'

const Home = () => {
  return (
    <div className={``}>
      <S.Container>
        <Heading>Teste</Heading>
      </S.Container>
    </div>
  )
}

export default Home
