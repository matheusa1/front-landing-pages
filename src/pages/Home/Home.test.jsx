import { screen } from '@testing-library/react'

import Home from '.'
import { renderTheme } from '../../styles/renderTheme'
import { theme } from '../../styles/theme'

test('renders learn react link', () => {
  renderTheme(<Home />)
  const headingContainer = screen.getByRole('heading', {
    name: 'Home',
  }).parentElement

  expect(headingContainer).toHaveStyle({
    background: theme.colors.mainBg,
  })

  expect(headingContainer).toMatchSnapshot()

  expect(headingContainer).toHaveStyleRule('background', '#F00')
})
