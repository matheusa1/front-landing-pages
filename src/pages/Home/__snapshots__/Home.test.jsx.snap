// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
.c0 {
  background: #F00;
}

<div
  class="c0"
>
  <h1>
    Home
  </h1>
</div>
`;
